py4web/core.py

```diff
@@ -953,20 +955,42 @@ def error404(error):
 # Web Server and Reload Logic: Operations
 #########################################################################################
 
+def watch_folder_event_loop(apps_folder):
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(loop)
+    loop.run_until_complete(watch_folder(apps_folder))
+
+async def watch_folder(apps_folder):
+    click.echo('watching python file changes in: %s' % apps_folder)
+    async for changes in awatch(os.path.join(apps_folder)):
+        for app in set(map(lambda p: p.relative_to(apps_folder).parts[0],
+                        filter(lambda p: p.suffix == '.py',
+                                map(lambda pair: pathlib.Path(pair[1]), changes)))):
+            Reloader.import_app(app)
 
```

Comment:
just one question;): 
It looks very javastriptish  - maybe it is related to `async` somehow?
it would be more pythonic something like:
```python
for app in set([p.relative_to(apps_folder).parts[0]  
        for p in [pathlib.Path(pair[1]) for pair in changes] 
        if p.suffix == '.py'])
    Reloader.import_app(app)
```

Reply:
Yeah I am more familiar with iterator based style than list comprehension :), unless this is a hard requirement to be pythonic, I think it works well?
Also, wouldn't this create two intermediate list?

Reply:
I think the syntax is clearer using list comprehension, I made a commit on that. Thank you!